routing/road_graph.cpp

```diff
@@ -154,14 +166,40 @@ IRoadGraph::RoadInfo::RoadInfo(bool bidirectional, double speedKMPH, initializer
     : m_points(points), m_speedKMPH(speedKMPH), m_bidirectional(bidirectional)
 {}
 
-// IRoadGraph::CrossEdgesLoader ------------------------------------------------
-
-IRoadGraph::CrossEdgesLoader::CrossEdgesLoader(m2::PointD const & cross, TEdgeVector & outgoingEdges)
-    : m_cross(cross), m_outgoingEdges(outgoingEdges)
+// IRoadGraph::CrossOutgoingLoader ---------------------------------------------
+void IRoadGraph::CrossOutgoingLoader::LoadEdge(FeatureID const & featureId,
```

Comment:
It was not so easy as it's in your sample because these four "ifs" have significant difference. 

Nevertheless let's consider the approach I've done in commit "	ICrossEdgesLoader::LoadEdges() refactoring.". Some copy-paste in LoadEdges is removed but the method become less readable. I guess removing copy-paste is more important in that case. What do you think?